lib/rust/ensogl/component/grid-view/src/lib.rs

```diff
@@ -525,6 +538,29 @@ impl<E: Entry> GridView<E> {
         let widget = Widget::new(app, frp, model, display_object);
         Self { widget }
     }
+
+    fn move_selection_or_emit_movement_out_of_grid_prevented_event(
```

Comment:
This is a super long name that is not understandable. I do not get what it does. I dont know what does "emit movement" means and what is "grid prevented event". Pleae find a better name and add documentation.